baremetal/metal3machine_manager.go

```diff
@@ -510,11 +510,11 @@ func (m *MachineManager) Delete(ctx context.Context) error {
 			host.Spec.UserData = nil
```

Comment:
@maelk what about the `UserData`, is it fine to delete the content of this field?

Reply:
that one is pretty much always present since it comes from CAPI. but we could put the same check there, it would not hurt